app/controllers/api/v2/users_controller.rb

```diff
@@ -26,11 +28,35 @@ def accept_terms
       def generate_and_send_magic_link
         skip_authorization
 
-        GenerateAndSendMagicLinkService.call(user: current_user)
+        # If user is not found current_user returns "HTTP Token: Access denied.\n"
+        current_user = authenticate_or_request_with_http_token do |token|
+          AuthenticationService.new(logger: Rails.logger).execute(token)
+        end
+
+        if current_user.is_a?(User)
+          GenerateAndSendMagicLinkService.call(user: current_user)
+        else
+          send_email_address_unrecognised_email
```

Comment:
does this present another  issue of either
* confirming the non existence of email
* generating emails to unknown recipients

Reply:
I'm not sure I understand sorry ... 
Are you asking whether we should actually be sending emails to unknown recepients?

Reply:
> Are you asking whether we should actually be sending emails to unknown recepients?
Yes as its confirming non existence as well as sending email to unknown recipients